### DISCUSSION TITLE
Camera settings

### DISCUSSION THREAD

**Discussion by DenisStadnichenko** [shogo4405/HaishinKit.kt#192] [Q&A]
Hi, can you tell me how to switch on the flash, change the zoom and make the image mirror image?
I can't understand the settings (
Thank you for your answer

**Comment by shogo4405** [comment]
No way. Thanks.

**Reply by DenisStadnichenko** [reply]
Hi
Can you make CameraManager in Camera2Source public, this will give you the ability to work with flash

I also think it would be useful to be able to get a session to be able to change zooms

thank you for your response

**Reply by shogo4405** [reply]
I don't have plans to make it maintainable, so you can just copy and paste the source code of Camera2Source. Customizing it in your project is the expected use case.

**Reply by bhavinatharva** [reply]
Hello @shogo4405 

can you please guide where i can change for flash light regarding changes , actually i tried but getting some exception like camera is already in use
